storage/src/main/java/tech/pegasys/teku/storage/server/kvstore/KvStoreDatabase.java

```diff
@@ -338,20 +337,17 @@ public void storeFinalizedState(BeaconState state, Bytes32 blockRoot) {
     try (final FinalizedUpdaterCommon updater = finalizedUpdater()) {
       updater.addFinalizedState(blockRoot, state);
 
-      if (state.getSlot().equals(GENESIS_SLOT)) {
-        updater.addFinalizedStateRoot(state.hashTreeRoot(), state.getSlot());
-      } else {
-        final Optional<BeaconState> maybeLastState =
-            getLatestAvailableFinalizedState(state.getSlot().minus(ONE));
-        maybeLastState.ifPresentOrElse(
-            lastState -> {
-              final StateRootRecorder recorder =
-                  new StateRootRecorder(
-                      lastState.getSlot().increment(), updater::addFinalizedStateRoot, spec);
-              recorder.acceptNextState(state);
-            },
-            () -> updater.addFinalizedStateRoot(state.hashTreeRoot(), state.getSlot()));
-      }
+      final Optional<BeaconState> maybeLastState =
+          getLatestAvailableFinalizedState(state.getSlot().minus(ONE));
```

Comment:
Just to be completely safe, I'd make this `minusMinZero(ONE)`. That way if we ever do get the genesis state attempting to be stored here, we'll still handle it fine, just wind up recording it's state root a second time but that's ok as the database will deduplicate it anyway.